slippymap/osmtileprovider_test.go

```diff
@@ -1,53 +1,36 @@
 package slippymap
 
-import "testing"
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
 
 func TestGetTileAddress(t *testing.T) {
 
-	TileProvider := &OSMTileProvider{}
-
-	// check correct a... url is returned
-	url, err := TileProvider.GetTileAddress(1, 2, 3)
-	if err != nil {
-		t.Error(err)
-	}
-	if url != "http://a.tile.openstreetmap.org/3/1/2.png" {
-		t.Errorf("Expected http://a.tile.openstreetmap.org/3/1/2.png, got: %s", url)
-	}
-
-	// check correct b... url is returned
-	url, err = TileProvider.GetTileAddress(1, 2, 3)
-	if err != nil {
-		t.Error(err)
-	}
-	if url != "http://b.tile.openstreetmap.org/3/1/2.png" {
-		t.Errorf("Expected http://b.tile.openstreetmap.org/3/1/2.png, got: %s", url)
-	}
-
-	// check correct c... url is returned
-	url, err = TileProvider.GetTileAddress(1, 2, 3)
-	if err != nil {
-		t.Error(err)
-	}
-	if url != "http://c.tile.openstreetmap.org/3/1/2.png" {
-		t.Errorf("Expected http://c.tile.openstreetmap.org/3/1/2.png, got: %s", url)
-	}
-
-	// check correct a... url is returned (it loops back to a after c)
-	url, err = TileProvider.GetTileAddress(1, 2, 3)
-	if err != nil {
-		t.Error(err)
-	}
-	if url != "http://a.tile.openstreetmap.org/3/1/2.png" {
-		t.Errorf("Expected http://a.tile.openstreetmap.org/3/1/2.png, got: %s", url)
-	}
-
-	// set invalid tile prefix to generate an error
-	TileProvider.osm_url_prefix = 4
-	url, err = TileProvider.GetTileAddress(1, 2, 3)
-	if err != nil {
-		// test passes
-	} else {
-		t.Error("Expected an error, got none.")
-	}
+	t.Run("GetTileAddress loops through CDNs correctly", func(t *testing.T) {
+		// -1 so that the first increment lands us up on CDN A. Doesn't really matter where it starts though
+		TileProvider := &OSMTileProvider{osm_url_prefix: -1}
+
+		// check correct a... url is returned
+		url, err := TileProvider.GetTileAddress(1, 2, 3)
+		require.NoError(t, err)
+		assert.Equal(t, "http://a.tile.openstreetmap.org/3/1/2.png", url)
```

Comment:
10/10 recommend using https://github.com/stretchr/testify for test assertions. A far cry from rspec, but still compacter and more expressive than the built in go testing stuff

did here as an example, ymmv, happy to conform to whatever if you'd rather not use a lib for testing

Reply:
I’m new to go so happy with your recommendation!